src/main/scala/com/mongodb/spark/rdd/partitioner/MongoSplitVectorPartitioner.scala

```diff
@@ -84,11 +88,12 @@ class MongoSplitVectorPartitioner extends MongoPartitioner {
     })
   }
 
-  private def createPartitions(partitionKey: String, result: BsonDocument, locations: Seq[String]): Array[MongoPartition] = {
+  private def createPartitions(partitionKey: String, result: BsonDocument, locations: Seq[String], minMaxKey: (BsonValue, BsonValue)): Array[MongoPartition] = {
     result.getDouble("ok").getValue match {
       case 1.0 =>
-        val rightHandPartitionBoundaries = result.get("splitKeys").asInstanceOf[util.List[BsonDocument]].asScala.map(_.get(partitionKey))
-        val partitions = PartitionerHelper.createPartitions(partitionKey, rightHandPartitionBoundaries, locations)
+        val splitKeys = result.get("splitKeys").asInstanceOf[util.List[BsonDocument]].asScala.map(_.get(partitionKey))
+        val rightHandBoundaries = minMaxKey._1 +: splitKeys :+ minMaxKey._2
+        val partitions = PartitionerHelper.createPartitions(partitionKey, rightHandBoundaries, locations, false)
```

Comment:
No need to call PartitionerHelper.setLastBoundaryToLessThanOrEqualTo? Would it be wrong to do so?  I ask because every other place that createPartitions is called, the next line is 

     if (!addMinMax) PartitionerHelper.setLastBoundaryToLessThanOrEqualTo(partitionKey, partitions)

and if PartitionerHelper.setLastBoundaryToLessThanOrEqualTo should always be called with !addMinMax, then the logic could just be moved into PartitionerHelper.createPartitions instead of having to be called separately.